flink-core/src/main/java/org/apache/flink/core/fs/SafetyNetCloseableRegistry.java

```diff
@@ -124,8 +124,9 @@ protected boolean doUnRegister(
 		return null != innerCloseable && closeableMap.remove(innerCloseable) != null;
 	}
 
-	@Override
-	public void close() throws IOException {
+	@VisibleForTesting
+	CloseableReaperThread closeInternally() throws IOException {
```

Comment:
Instead of introducing this method, couldn't we use the `SafetyNetCloseableRegistry(Supplier<CloseableReaperThread> reaperThreadSupplier)` constructor to pass in a thread over which we hold control? I think this would be a bit cleaner.